scripts/auto_archive.py

```diff
@@ -0,0 +1,202 @@
+        # The program is not stopped since GitHub allows 60 query per hours with
+        # authentication. However the program will most likely fail.
+        logger.critical(e)
+
+    logger.info(f"Retrieving modfied datasets since {since}")
+    repo = Github(gh_access_token).get_repo("CONP-PCNO/conp-dataset")
+    commits = repo.get_commits(since=since, until=until)
+
+    with open(".conp-archive", "w") as fout:
+        fout.write(now.isoformat())
+
+    modified_datasets: set[str] = {
+        file_.filename
+        for commit in commits
+        for file_ in commit.files
+        if file_.filename.startswith("projects/")
+    }
+
+    return modified_datasets
+
+
+def archive_dataset(dataset_path: str, out_dir: str, version: str) -> None:
+    os.makedirs(os.path.dirname(out_dir), mode=0o755, exist_ok=True)
+    out_filename = f"{out_dir}_version-{version}.tar.gz"
+    logger.info(f"Archiving dataset: {dataset_path} to {out_filename}")
+
+    cwd = os.getcwd()
+    try:
+        datalad_archiver = export_archive.ExportArchive()
+        dataset_repo = git.Repo(dataset_path)
+
+        with open(os.path.join(dataset_path, ".git.log"), "w") as fout:
+            fout.write(dataset_repo.git.log(pretty="format:%H %s"))
+
+        # Export is performed from the dataset root.
+        # This is to avoid failure when a submodule is not downloaded; e.g. for parent
+        # dataset in dataset derivative.
+        os.chdir(os.path.join(cwd, dataset_path))
+        datalad_archiver(".", filename=out_filename)
+
+    except Exception as e:
+        raise ArchiveFailed(
+            f"FAILURE: could not archive dataset: {dataset_path} to {out_filename}\n{e}"
+        )
+    finally:
+        os.chdir(cwd)
+
+
+if __name__ == "__main__":
+    args = parse_args()
+    datasets = get_all_datasets() if args.all else get_modified_datasets()
+
+    for dataset in datasets:
+        dataset_name = dataset.removeprefix("projects/")
+
+        try:
+            logger.info(f"Installing dataset: {dataset}")
+            install_dataset(dataset)
+
+            is_public = False
+            version = ""
+            dataset_size = 0.0
+
+            with open(os.path.join(dataset, "DATS.json")) as fin:
+                metadata = json.load(fin)
+
+                is_public = metadata.get("privacy") == "open"
+                version = metadata.get("version")
+
+                for distribution in metadata.get("distributions", list()):
+                    dataset_size += humanfriendly.parse_size(
+                        f"{distribution['size']} {distribution['unit']['value']}",
+                    )
+                    dataset_size //= 1024 ** 3  # Convert to GB
+
+            # Only archive public dataset less than 20GB
+            if dataset_size <= args.max_size and is_public:
+                logger.info(f"Downloading dataset: {dataset}")
+                get_dataset(dataset)
+                for submodule in get_proper_submodules(dataset):
```

Comment:
I am guessing here that you are reusing the same function that is used for the tests to identify whether a submodule is a derived dataset or actually part of the dataset?

Reply:
Yes this reuses the same function.